cypress/integration/post-gallery.js

```diff
@@ -0,0 +1,85 @@
+/// <reference types="Cypress" />
+import { userFactory } from '../fixtures/user';
+
+describe('Post Gallery Block', () => {
+  beforeEach(() => cy.configureMocks());
+
+  /** @test */
+  it('Displays user name for a non anonymous post', () => {
+    const user = userFactory();
+
+    cy.mockGraphqlOp('ContentfulBlockQuery', {
+      block: {
+        __typename: 'PostGalleryBlock',
+      },
+    });
+
+    cy.mockGraphqlOp('ActionGalleryQuery', {
+      posts: [
+        {
+          actionDetails: {
+            anonymous: false,
+          },
+          user: {
+            firstName: user.firstName,
+          },
+        },
+      ],
+    });
+
+    cy.authVisitBlockPermalink(user, '123');
+
+    cy.findByTestId('post-author').should('have.length', 1);
+  });
+
+  /** @test */
+  it('Does not display user name for a non anonymous post', () => {
```

Comment:
I think this should be anonymous!

```suggestion
  it('Does not display user name for an anonymous post', () => {
```

Reply:
OH WAIT this is in the profile right? So we DO want to see the name?

Reply:
My understanding is that we're testing here that for anonymous posts, the username does _not_ display to ensure anonymity? (This would be in a post gallery which are rendered via a contentful block but I think this logic applies globally to all post cards cross site).

Reply:
Omg I read your suggestion as the actual code! Yes indeed, thank! I'll commit this in. (This was all a copy pasta from the original so must have had a typo.